src/pages/GanacheDetails.ts

```diff
@@ -0,0 +1,370 @@
+    const batchRequest = web3.createBatchRequest();
+
+    // Adds the getBlock action to the request
+    blockNumbers.forEach((block) => {
+      batchRequest.add(web3.eth.getBlock, block, true);
+    });
+
+    // Retrieves all blocks from batch execution
+    const blocks: any[] = await batchRequest.execute();
+
+    // Retrieves all accounts from the we3 provider
+    const accounts: string[] = await web3.eth.getAccounts();
+
+    // Retrieves all transactions from the we3 provider filtering by account
+    // **** Account filter is required because Ganache can be launched with fork options enabled)
+    const txs: TTxs[] = [];
+    await Promise.all(
+      blocks.map(async (block) => {
+        const transactions: any = Object.values(block.transactions).filter((transaction: any) =>
+          accounts.includes(transaction.from)
+        );
+
+        await Promise.all(
+          Object.values(transactions).map(async (transaction: any) => {
+            const receipt = await web3.eth.getTransactionReceipt(transaction.hash);
+
+            const tx: TTxs = {
+              hash: transaction.hash,
+              value: transaction.value,
+              fromAddress: receipt.from,
+              gasUsed: receipt.gasUsed,
+              createdContractAddress: receipt.contractAddress,
+            };
+
+            txs.push(tx);
+          })
+        );
+      })
+    );
+
+    // Returns the filtered transactions
+    return txs;
+  }
+
+  /**
+   * This function is responsible for retrieving the html from within the file
+   * and returning it to the web provider so that it can be printed.
+   */
+  private async getHtmlForWebview(): Promise<string> {
+    // Retrieves the HTML from the file
+    const file = await fs.readFile(
+      this._context.asAbsolutePath(path.join('resources', 'ganache', 'index.html')),
+      'utf8'
+    );
+
+    // Creates a virtual path so that assets from within the HTML can be loaded and return the HTML
+    const virtualPath = this._panel?.webview.asWebviewUri(this._context.extensionUri).toString();
+    return file.replace(/{{root}}/g, virtualPath!);
+  }
+}
+
+/**
+ * This function is responsible for registering the Ganache Details panel
+ *
+ * @param context - An extension context is a collection of utilities private to an extension
+ */
+export async function registerGanacheDetails(context: vscode.ExtensionContext): Promise<void> {
+  // Creates the webview instance
+  const ganacheDetails = new GanacheDetails(context);
+
+  // Registers the command responsible for creating the WebviewPanel
+  context.subscriptions.push(
+    vscode.commands.registerCommand('truffle-vscode.getGanacheServerInfo', async (projectView: ProjectView) => {
+      // Creates or show the WebviewPanel
+      const panel = await ganacheDetails.createOrShowWebviewPanel(projectView);
+
+      // Once the Webview Panel has been created it will be added to subscriptions
+      if (panel) context.subscriptions.push(panel);
```

Comment:
If I have 2 Ganache services, the details tab will be reused across services, is this intended behavior? Maybe would be better to display details in separate tabs, if it is not too difficult.

Reply:
Yes @acuarica, you're right, I tried to do that but it started to give a lot of trouble when I needed to switch among the tabs. So I decided to leave it that way in this first version so as not to waste too much time. But we can open a new issue to investigate how to open more than one instance for the same component. I will open it and feel free to help me on that :)